src/test/unit/plugin_tests/table_test.js

```diff
@@ -905,120 +905,85 @@ test("Deleting all columns removes table", function () {
     });
 });
 
-test("Deselects before removing row that contains collapsed selection",
-     function () {
+test("Removing tr, caret is set to first td of previous tr if no next tr", function () {
+    var prevTd;
     manipulationTestHelper({
         startHtml: basicTableHtml,
         setCaretInSelector: "#td_3_1",
         manipulationFunc: function (wymeditor) {
             var tr = wymeditor.$body().find("#tr_3")[0];
+            prevTd = tr.previousSibling.childNodes[0];
+            if (jQuery.browser.msie && jQuery.browser.versionNumber === 8) {
+                // the caret ends up inside #span_2_1,
+                // which is fine
+                prevTd = prevTd.childNodes[0];
+            }
             wymeditor.tableEditor.removeRow(tr);
         },
         additionalAssertionsFunc: function (wymeditor) {
             expectOneMore();
-            // This doesn't really test that .deselect() was called. We'd need
-            // a spy for that. Good enough.
             strictEqual(
-                wymeditor.hasSelection(),
-                false,
-                "No selection"
+                wymeditor.selectedContainer(),
+                prevTd,
+                "selected container"
             );
-        },
-        expectedResultHtml: removedRow3Html
```

Comment:
I liked also checking the HTML because it ensures that we didn't screw up the test setup and the row was actually removed. There's a purity for pragmatism trade-off, there, but your call.